src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  getArea: () => number;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-export class Circle {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side\'s length has to be greater than 0');
+    } else if (
+      (a > b && a > c && a >= b + c)
+      || (b > a && b > c && b >= a + c)
+      || (c > a && c > b && c >= a + b)
+    ) {
+      throw new Error('The longes side of a triangle cannot be'
+        + 'greater or equal than a sum of two others');
+    }
 
+    this.getArea = function getArea(): number {
+      const semiperimeter = 0.5 * (this.a + this.b + this.c);
+
+      return +Math.sqrt(semiperimeter * (semiperimeter - this.a)
+        * (semiperimeter - this.b) * (semiperimeter - this.c)).toFixed(2);
+    };
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  getArea: () => number;
+
+  constructor(
+    public color: Color,
+    public radius: number,
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again